pnet_datalink/src/pcap.rs

```diff
@@ -45,7 +47,11 @@ impl Default for Config {
 /// Create a datalink channel from the provided pcap device
 #[inline]
 pub fn channel(network_interface: &NetworkInterface,
-               config: Config) -> io::Result<datalink::Channel> {
+               config: Config) -> io::Result<super::Channel> {
+    if config.read_buffer_size < 65516 {
+        return Err(io::Error::new(io::ErrorKind::Other, "read_buffer_size too small"));
```

Comment:
If this explicit check is not in place, what happens then? If possible I usually prefer if the underlying implementation returns errors when something is wrong. Partly because if they have a similar check then this is just duplication of their code. Partly because if the underlying library changes this requirement then we would not also need a change here to be compatible with libpcap again.

Reply:
Iff we need the check here, then I think the magic number should be moved to a named static at the start of the file.

Reply:
There is an error reported from pcap, but it's non-obvious:
> Activate: can't mmap rx ring: Invalid argument
One also can't distinguish between a "runtime" mmap failure (e.g. OOM) and a too-small buffer. The minimum is actually a kernel limitation, so you're right, we should not hard-code it here.

Probably the most user-friendly thing to do would be to separately wrap the results from `buffer_size`, so that we can return a more meaningful error.

Reply:
Ah, looks like the logic is trivial; the minimum size basically just MAXIMUM_SNAPLEN. Should be easy and safe to validate here.

Reply:
Can you get that constant from pcap? I still feel it's sketchy to code a constant here to perform some input validation that should be the responsibility of pcap. I would prefer not checking here and maybe add a section in the docs for that buffer variable that it must be bigger than MAXIMUM_SNAPLEN or something?

With invalid config data a lot of the backends produce very strange errors. I don't really see it as our responsibility to fix that here. Then better to make a PR to those libraries and make their error messages better.

What do you think @mrmonday?

Reply:
I agree with @faern - I would rather use the backend's errors. If there's particular setup we're meant to be doing but aren't, that's a different matter.

Of course, there are two scenarios here:

 * Using backend specific configuration
 * Using backend-agnostic configuration

In the first case, we should always pass through the exact values to the backend.

In the second case, I think it can be argued either way. Backends are free to ignore or use values that are applicable. I don't like silently changing values, but I think I'd be ok with detecting errors before we hit the backend.

Reply:
I'm happy to restore the original behavior; one change that would be helpful going forward was a better way of merging the `lib::Config::Default` and individual backend defaults.

In the previous versions of this change, when I had `pcap` acting as the default backend, the default configuration was broken, because the `lib::Config::Default.read_buffer_size` will never work, with any foreseeable version of pcap (the snaplen will certainly never get _smaller_...). So it's an obvious condition that is trivially correctable.

My original addition of this fix was in the implementation of `From<super::Config>` for pcap. I think that's probably the right place if anywhere.